.github/actions/check-version-bumped/action.yml

```diff
@@ -0,0 +1,20 @@
+---
+name: Check if version bumped
+description: Checks if version in Cargo.toml is bumped
+
+inputs:
+  cargo-toml-path:
+    description: "Path to Cargo.toml"
+    type: string
+    required: true
+
+runs:
+  using: composite
+  steps:
+    - name: Check
+      run: |
+        if ! git diff HEAD origin/main -- ${{ inputs.cargo-toml-path }} | grep -q '^+version ='; then
```

Comment:
It's better to escape paths with double quotes `"` when dealing with paths in scripts, as at some point they might contain spaces, so I suggest here `"${{ inputs.cargo-toml-path }}"`